src/components/PeopleFilters.tsx

```diff
@@ -1,96 +1,120 @@
+import { Link, useSearchParams } from 'react-router-dom';
+import { ChangeEventHandler } from 'react';
+import { getSearchWith, SearchParams } from '../utils/searchHelper';
+import { CenturiesFilter } from '../types/CenturiesFilter';
+import classNames from 'classnames';
+import { SexFilter } from '../types/SexFilter';
+
 export const PeopleFilters = () => {
-  return (
-    <nav className="panel">
-      <p className="panel-heading">Filters</p>
+  const [searchParams, setSearchParams] = useSearchParams();
 
-      <p className="panel-tabs" data-cy="SexFilter">
-        <a className="is-active" href="#/people">
-          All
-        </a>
-        <a className="" href="#/people?sex=m">
-          Male
-        </a>
-        <a className="" href="#/people?sex=f">
-          Female
-        </a>
-      </p>
+  const setSearchWith = (paramsToUpdate: SearchParams) => {
+    const search = getSearchWith(searchParams, paramsToUpdate);
 
-      <div className="panel-block">
-        <p className="control has-icons-left">
-          <input
-            data-cy="NameFilter"
-            type="search"
-            className="input"
-            placeholder="Search"
-          />
+    setSearchParams(search);
+  };
 
-          <span className="icon is-left">
-            <i className="fas fa-search" aria-hidden="true" />
-          </span>
-        </p>
-      </div>
+  const sex = searchParams.get('sex');
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries');
 
-      <div className="panel-block">
-        <div className="level is-flex-grow-1 is-mobile" data-cy="CenturyFilter">
-          <div className="level-left">
-            <a
-              data-cy="century"
-              className="button mr-1"
-              href="#/people?centuries=16"
-            >
-              16
-            </a>
+  const queryHandler: ChangeEventHandler<HTMLInputElement> = e => {
```

Comment:
It's a good practice to use useCallback for event handlers. It will prevent unnecessary re-renders of the component. Here, every time the component re-renders, a new function is created for queryHandler. You can wrap it with useCallback to memoize the function.